test/staking/DonationsStaking.test.ts

```diff
@@ -255,15 +257,37 @@ describe("DonationsStaking - DonationStaking contract that receives funds in ETH
     const avatarDaiBalanceBeforeEnd = await dai.balanceOf(avatar);
     let isActive = await donationsStaking.active();
     expect(isActive).to.be.equal(true);
+    const ethBalanceBeforeWithdraw = await donationsStaking.provider.getBalance(
+      donationsStaking.address
+    );
+    const avatarETHBalanceBeforeWithdraw =
+      await donationsStaking.provider.getBalance(avatar);
+    const balance = await goodCompoundStaking.balanceOf(
+      donationsStaking.address
+    );
+
     const encoded = donationsStaking.interface.encodeFunctionData("withdraw");
     await genericCall(donationsStaking.address, encoded);
 
     isActive = await donationsStaking.active();
+
     const totalStakedAfterEnd = await donationsStaking.totalStaked();
     const avatarDaiBalanceAfterEnd = await dai.balanceOf(avatar);
     expect(avatarDaiBalanceAfterEnd).to.be.gt(avatarDaiBalanceBeforeEnd);
     expect(avatarDaiBalanceAfterEnd).to.be.equal(totalStakedBeforeEnd);
+    expect(avatarDaiBalanceAfterEnd).to.be.equal(
+      avatarDaiBalanceBeforeEnd.add(balance)
+    );
+    const ethBalanceAfterWithdraw = await donationsStaking.provider.getBalance(
+      donationsStaking.address
+    );
+    const avatarETHBalanceAfterWithdraw =
+      await donationsStaking.provider.getBalance(avatar);
     expect(totalStakedAfterEnd).to.be.equal(0);
+    expect(ethBalanceAfterWithdraw).to.be.equal(0);
+    expect(avatarETHBalanceAfterWithdraw).to.be.equal(
+      ethBalanceBeforeWithdraw.add(avatarETHBalanceBeforeWithdraw)
+    );
```

Comment:
not good and incorect
1. you do not verify values are not trivial ie 0. if you dont use hardcoded numbers you have to verify values are not trivial!
2. eth should STAY at staking, only tokens not sold because of 0.3% should be transfered to avatar

Reply:
what do you mean even in withdraw, JUST in withdraw we transfer tokens back to avatar